### DISCUSSION TITLE
s390x -> ./pocketbase: cannot execute binary file: Exec format error

### DISCUSSION THREAD

**Discussion by Magnamura** [pocketbase/pocketbase#572] [Q&A] [answered]
Hey.
I can't start pocketbase on my (vm) System.
`Operating System: Debian GNU/Linux 11 (bullseye)
 Kernel: Linux 5.10.0-17-s390x
 Architecture: s390x`

Any plans on implementing s390x support?

**Comment by ganigeorgiev** [comment, marked as answer]
PocketBase uses by default `modernc.org/sqlite` (CGO-free port of SQLite) which currently doesn't support s390x as build target - https://pkg.go.dev/modernc.org/sqlite#hdr-Supported_platforms_and_architectures.

But you can build the base PocketBase app on your own with the integrated CGO driver (mattn/go-sqlite3):

1. Clone the repo and navigate to `examples/base` or create a Go file as [shown in the README](https://github.com/pocketbase/pocketbase#example).

2. Cross compile with `CGO_ENABLED=1` and `GOARCH=s390x`:
    ```sh
    CGO_ENABLED=1 GOOS=linux GOARCH=s390x go build
    ```
    _(you can view all possible goos and goarch values here - https://go.dev/doc/install/source#environment)_


_I'll move the issue to Q&A discussion._
